src/hooks/useRepayLoanHandler.tsx

```diff
@@ -1,24 +1,129 @@
-import {useState} from 'react'
+import {useState, useContext, useEffect} from 'react'
+import { ethers, Contract } from 'ethers'
+import { WalletContext } from '../context/WalletContext'
+import PosManager from '../../abis/v1-periphery/PositionManager.sol/PositionManager.json'
+import GammaPool from '../../abis/v1-core/GammaPool.sol/GammaPool.json'
+import toast from 'react-hot-toast'
 
 export const useRepayLoanHandler = () => {
+  const { accountInfo, provider, connectWallet } = useContext(WalletContext)
   const [repayAmt, setRepayAmt] = useState<number>(0)
   const [enableRepay, setEnableRepay] = useState<Boolean>(false)
+  const [posManager, setPosManager] = useState<Contract | null>(null)
+  const [gammaPool, setGammaPool] = useState<Contract | null>(null)
+  const [tokenId, setTokenId] = useState(19)
+  const [loanAmount, setLoanAmount] = useState<number>(0)
+  const [repayCal, setRepayCal] = useState(0)
+  const [cfmm, setCfmm] = useState()
   let percentages = [25, 50, 75, 100]
 
-  function changeSliderPercentage(value: number) {
+  useEffect(() => {
+    async function fetchContract() {
+      if (!provider) {
+        console.log('Please connect wallet.')
+        return
+      }
+
+      // Position Manager contract address
+      let address = process.env.NEXT_PUBLIC_POSITION_MANAGER_ADDRESS || ''
+      let gammaPoolAddress = process.env.NEXT_PUBLIC_GAMMAPOOL_ADDRESS || ''
+
+      // Variable to hold Position Manager contract
+      let _posManager = null
+      let _gammaPool
+
+      _gammaPool = new ethers.Contract(
+        gammaPoolAddress,
+        GammaPool.abi,
+        accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+      )
+      if (_gammaPool) {
+        setGammaPool(_gammaPool)
+      }
+
+      _posManager = new ethers.Contract(
+        address,
+        PosManager.abi,
+        accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+      )
+      if (_posManager) {
+        setPosManager(_posManager)
+      }
+    }
+    fetchContract()
+  }, [provider])
```

Comment:
Doesn't the provider change only when the user connects with metamask? What happens if the user does not gone to the repay loan page yet while he is already connected? When will these contracts get set?

Reply:
Provider changes when page loads. Because of WalletContext.